src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,72 @@
 'use strict';
 
+const NAME = '';
+const HEALTH = 100;
+
 class Animal {
-  // write your code here
+  static alive = [];
+
+  health;
+  name;
+
+  constructor(name = NAME, health = HEALTH) {
+    this.name = name;
+    this.health = health;
+
+    if (this.health > 0) {
+      Animal.alive.push(this);
+    }
+  }
+
+  takeDamage(damage) {
+    this.health -= damage;
+
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
 }
 
+const HIDDEN = false;
+
 class Herbivore extends Animal {
-  // write your code here
+  hidden;
+
+  constructor(name, health, hidden = HIDDEN) {
+    super(name, health);
+
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
+
+  takeDamage(damage) {
```

Comment:
The `takeDamage` method should include logic to remove the animal from the `Animal.alive` array if its health drops below or equal to 0. This is currently handled in the `Animal` class, but since you're overriding `takeDamage` here, you need to ensure the logic is applied.